### PR TITLE
Emballage fee

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
This PR includes:
1. Added emballage price during product import to product metadata
2. Added emballage fee on woocommerce cart
3. Unit test

### CODE REVIEW

**Comment by jnmrclmbsjse** [comment]
Task linked: [CU-34c5jy0 Woocomerce embalage (required addons)](https://app.clickup.com/t/34c5jy0)

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        $items = $woocommerce->cart->get_cart();
+
+        $totalEmballagePrice = 0.00;
+        foreach ($items as $values) {
+            $product = wc_get_product($values['product_id']);
+            if ($product) {
+                $emballagePrice = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY);
+                if ($emballagePrice) {
+                    $totalEmballagePrice += (float) $emballagePrice;
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php:24]: it should be times the qty

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                }
+            }
+        }
+
+        if ($totalEmballagePrice > 0) {
+            $woocommerce->cart->add_fee(__('Emballage fee', I18N::DOMAIN), $totalEmballagePrice);
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php:30]: does it get removed if the you remove the product on cart? or only calculated on checkout

**Review by ikulis — COMMENTED** [review]
The emballage fee needs to be send with correct tax_rate_id to backend, can you also add a unit test for it? 

like having the order
* 2x product1 with emaballage 3.33
* 1x product2 with emaballage 1.11

order to backend should be send with total fee 7.77 and correct tax_rate_id.

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php (hunk 1 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php]: Added quantity multiplier.

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php (hunk 2 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php]: Yes, fees added using `add_fee` does not persist until order is created on checkout.

File: src/StoreKeeper/WooCommerce/B2C/Factories/LoggerFactory.php [hunk 3]
```diff
@@ -29,7 +29,7 @@ public static function getWpLogDirectory($suffix = ''): ?string
             $log_dir .= DIRECTORY_SEPARATOR.$suffix;
         }
         if (!file_exists($log_dir)) {
-            mkdir($log_dir, 0777, true);
+//            mkdir($log_dir, 0777, true);
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Factories/LoggerFactory.php:32]: ?

File: src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php [hunk 4]
```diff
@@ -20,6 +20,9 @@
 
 class OrderExport extends AbstractExport
 {
+    const EMBALLAGE_TAX_RATE_ID_META_KEY = 'storekeeper_emballage_tax_id';
+    const IS_EMBALLAGE_FEE_KEY = 'is_emballage_fee';
+    const TAX_RATE_ID_FEE_KEY = 'sk_tax_rate_id';
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php:25]:
```suggestion
    const EMBALLAGE_TAX_RATE_ID_META_KEY = 'sk_emballage_tax_id';
    const IS_EMBALLAGE_FEE_KEY = 'sk_is_emballage_fee';
    const TAX_RATE_ID_FEE_KEY = 'sk_tax_rate_id';
```

Let's have same prefix for all

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php [hunk 5]
```diff
@@ -15,19 +16,33 @@ public function addEmballageFee()
...
         foreach ($items as $values) {
             $product = wc_get_product($values['product_id']);
+            $quantity = $values['quantity'];
             if ($product) {
-                $emballagePrice = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY);
-                if ($emballagePrice) {
-                    $totalEmballagePrice += (float) $emballagePrice;
+                if ($product->meta_exists(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY)) {
+                    $emballagePrice = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY);
+                    $totalEmballagePrice += (float) ($emballagePrice * $quantity);
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php:27]:
```suggestion
                    $totalEmballagePrice +=  round($emballagePrice*100) * $quantity;
```
We could really count in in cents to not have any rounding errors

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+{
+    public function addEmballageFee()
+    {
+        /* @var WooCommerce $woocommerce */
+        global $woocommerce;
+
+        $items = $woocommerce->cart->get_cart();
+
+        $lastEmballageTaxRateId = null;
+        $totalEmballagePrice = 0.00;
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php:20]:
```suggestion
        $totalEmballagePriceInCents = 0;
```

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                if ($product->meta_exists(ProductImport::PRODUCT_EMBALLAGE_TAX_ID_META_KEY)) {
+                    $lastEmballageTaxRateId = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_TAX_ID_META_KEY);
+                }
+            }
+        }
+
+        if ($totalEmballagePrice > 0) {
+            $emballagePrice = [
+                'name' => __('Emballage fee', I18N::DOMAIN),
+                'amount' => $totalEmballagePrice,
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php:39]:
```suggestion
                'amount' => round($totalEmballagePriceInCents/100,2) ;
```

File: src/StoreKeeper/WooCommerce/B2C/Factories/LoggerFactory.php (hunk 3 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Factories/LoggerFactory.php]: Reverted commented code.

File: src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php (hunk 4 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php]: Changes implemented.

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php (hunk 5 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php]: Changes implemented.

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php (hunk 6 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php]: Changes implemented.

File: src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php (hunk 7 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php:40]: Changes implemented.

**Pull request merged**